e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/report-reconciliation/bank-utilisation-page.spec.js

```diff
@@ -0,0 +1,141 @@
+import {
+  CURRENCY,
+  FeeRecordEntityMockBuilder,
+  PaymentEntityMockBuilder,
+  REPORT_NOT_RECEIVED,
+  RECONCILIATION_IN_PROGRESS,
+  UtilisationReportEntityMockBuilder,
+  toIsoMonthStamp,
+  toMonthYearString,
+  getPreviousReportPeriodForBankScheduleByMonth,
+  FEE_RECORD_STATUS,
+} from '@ukef/dtfs2-common';
+import pages from '../../../pages';
+import USERS from '../../../../fixtures/users';
+import { NODE_TASKS } from '../../../../../../e2e-fixtures';
+
+const { utilisationReportPage } = pages;
+
+context('Bank utilisation report page', () => {
+  const SUBMISSION_MONTH = toIsoMonthStamp(new Date());
+  const BANK_ID = '961';
+  const REPORT_ID = 1;
+  const FEE_RECORD_ID_ONE = '11';
+  const FEE_RECORD_ID_TWO = '22';
+  const FACILITY_ID_ONE = '11111111';
+  const FACILITY_ID_TWO = '22222222';
+  let bankName;
+  let reportPeriodString;
+
+  const aTfmFacilityFacilitySnapshot = () => ({
+    coverStartDate: new Date().getTime(),
+    coverEndDate: new Date().getTime(),
+    interestPercentage: 5,
+    dayCountBasis: 365,
+    coverPercentage: 80,
+    value: 100000,
+  });
+
+  const TFM_FACILITIES = [
+    {
+      facilitySnapshot: { ...aTfmFacilityFacilitySnapshot(), ukefFacilityId: FACILITY_ID_ONE },
+    },
+    {
+      facilitySnapshot: { ...aTfmFacilityFacilitySnapshot(), ukefFacilityId: FACILITY_ID_TWO },
+    },
+  ];
+
+  before(() => {
+    const visibleBanks = [];
+
+    cy.task(NODE_TASKS.DELETE_ALL_TFM_FACILITIES_FROM_DB);
+    cy.task(NODE_TASKS.INSERT_TFM_FACILITIES_INTO_DB, TFM_FACILITIES);
+
+    cy.task(NODE_TASKS.GET_ALL_BANKS).then((getAllBanksResult) => {
+      getAllBanksResult
+        .filter((bank) => bank.isVisibleInTfmUtilisationReports)
+        .forEach((visibleBank) => {
+          visibleBanks.push(visibleBank);
+        });
+    });
+
+    cy.task(NODE_TASKS.DELETE_ALL_FROM_SQL_DB);
+
+    cy.wrap(visibleBanks).each((bank) => {
+      const reportPeriod = getPreviousReportPeriodForBankScheduleByMonth(bank.utilisationReportPeriodSchedule, SUBMISSION_MONTH);
+
+      reportPeriodString = toMonthYearString(reportPeriod.start);
```

Comment:
Please can we use the existing report period formatters here? If the report was quarterly this would not be just a month and year and if it is monthly than the existing formatter will return the month and year.

Reply:
Where can I find this formatter?

Reply:
done